airgun/views/contentview.py

```diff
@@ -210,3 +326,123 @@ def is_displayed(self):
+                and self.next.is_displayed
+        )
+
+
+class ContentViewVersionRemoveConfirmationView(BaseLoggedInView):
+    breadcrumb = BreadCrumb()
+
+    cancel = Text(".//button[normalize-space(.)='Cancel']")
+    back = Text(".//button[@ng-click='transitionBack()']")
+    confirm_remove = Text(".//button[@ng-click='performDeletion()']")
+
+    @property
+    def is_displayed(self):
+        breadcrumb_loaded = self.browser.wait_for_element(
+            self.breadcrumb, exception=False)
+        return (
+                breadcrumb_loaded
+                and self.breadcrumb.locations[0] == 'Content Views'
+                and len(self.breadcrumb.locations) == 3
+                and self.breadcrumb.read() == 'Deletion'
+                and self.confirm_remove.is_displayed
+        )
+
+
+class CreateYumFilterView(BaseLoggedInView):
+    breadcrumb = BreadCrumb()
+
+    name = TextInput(id='name')
+    content_type = Select(id='type')
+    inclusion_type = Select(id='inclusion')
+    description = TextInput(id='description')
+
+    save = Text('.//button[contains(@ng-click, "handleSave()")]')
+    cancel = Text('.//button[@ng-click="handleCancel()"]')
+
+    @property
+    def is_displayed(self):
+        breadcrumb_loaded = self.browser.wait_for_element(
+            self.breadcrumb, exception=False)
+        return (
+            breadcrumb_loaded
+            and self.breadcrumb.locations[0] == 'Content Views'
+            and self.breadcrumb.read() == 'Create Yum Filter'
+        )
+
+
+class EditYumFilterView(BaseLoggedInView):
+    breadcrumb = BreadCrumb()
+
+    @property
+    def is_displayed(self):
+        breadcrumb_loaded = self.browser.wait_for_element(
+            self.breadcrumb, exception=False)
+        return (
+            breadcrumb_loaded
+            and len(self.breadcrumb.locations) > 3
+            and self.breadcrumb.locations[2] == 'Yum Filters'
+            and self.breadcrumb.read() != 'Create Yum Filter'
+        )
+
+    @View.nested
+    class details(SatSecondaryTab):
+        name = EditableEntry(name='Name')
+        description = EditableEntry(name='Description')
+
+    @View.nested
+    class rpms(SatSecondaryTab, SearchableViewMixin):
+        TAB_NAME = 'RPMs'
+
+        exclude_no_errata = Checkbox(
+            locator=".//input[@type='checkbox']"
+                    "[@ng-model='filter.original_packages']"
+        )
+        add_rule = Text(".//button[@ng-click='addRule()']")
+        remove_rule = Text(".//button[@ng-click='removeRules(filter)']")
+
+        table = SatTable(
+            locator='//table',
+            column_widgets={
+                0: Checkbox(locator=".//input[@type='checkbox']"),
+                4: Text("./button[contains(@ng-click, 'rule.editMode')]"),
+            },
+        )
+
+    @View.nested
+    class affected_repositories(SatSecondaryTab):
+        TAB_NAME = 'Affected Repositories'
+        filter_toggle = RadioGroup(".//div[@class='col-sm-8']")
```

Comment:
any other alternatives

Reply:
That page isn't that nice, and that's just a start :)
`<div class="col-sm-8">` is all i have for that `div`. I can find radio button and go 3 parents higher but i'm not sure such locator will be way more readable, smth like that:
`//div[div/label/input[@ng-model='showRepos']]` (can't use `//` there to shorten path as those div elements are inside other divs, so many results will be found)

Reply:
okay